### PR TITLE
feat(logger): add SwanLabLogger for enhanced logging capabilities

### PR DESCRIPTION
**Pull request by relic-yuexi**
## What does this PR do?

This PR adds a new `SwanLabLogger` to PyTorch Lightning, providing integration with SwanLab (https://swanlab.cn) for enhanced experiment tracking and logging capabilities.

### Key Features:
- **Metrics Logging**: Automatically log training/validation/test metrics during training
- **Hyperparameter Tracking**: Save model hyperparameters and configuration
- **Media Logging Support**: Log images, text, audio, and video files
- **Model Checkpoint Logging**: Optional automatic logging of model checkpoints
- **Flexible Modes**: Support for cloud, local, offline, and disabled modes
- **Experiment Management**: Create and manage SwanLab experiments with custom project names and descriptions

### Implementation Details:
- Created `src/lightning/pytorch/loggers/swanlab.py` with full `SwanLabLogger` implementation
- Updated `src/lightning/pytorch/loggers/__init__.py` to export `SwanLabLogger`
- Follows existing Lightning logger patterns and conventions
- Compatible with `swanlab>=0.3.0`

### Dependencies:
- Requires `swanlab>=0.3.0` package (optional dependency)

### Usage Example:

```python
from lightning.pytorch.loggers import SwanLabLogger
from lightning.pytorch import Trainer

# Create logger
swanlab_logger = SwanLabLogger(
    project="my_project",
    experiment_name="experiment_1",
    description="My experiment description"
)

# Use with Trainer
trainer = Trainer(logger=swanlab_logger)
```

### Before submitting

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21506.org.readthedocs.build/en/21506/

### CODE REVIEW

**Comment by bhimrazy** [comment]
Hi @relic-yuexi, thanks for the contribution!
Closing this PR as per the comment: https://github.com/Lightning-AI/pytorch-lightning/pull/21584#issuecomment-4057894534

image

cc: @justusschock
